src/storage/db_interface_common.py

```diff
@@ -178,12 +179,22 @@ def _extract_binaries(self, analysis_dict, key, uid):
         for analysis_key in analysis_dict[key].keys():
             if analysis_key != 'summary':
                 file_name = '{}_{}_{}'.format(get_safe_name(key), get_safe_name(analysis_key), uid)
-                self.sanitize_fs.put(pickle.dumps(analysis_dict[key][analysis_key]), filename=file_name)
+                self._store_in_sanitize_db(pickle.dumps(analysis_dict[key][analysis_key]), file_name)
                 tmp_dict[analysis_key] = file_name
             else:
                 tmp_dict[analysis_key] = analysis_dict[key][analysis_key]
         return tmp_dict
 
+    def _store_in_sanitize_db(self, content: bytes, file_name: str):
+        if self.sanitize_fs.exists({'filename': file_name}):
+            md5_hash = md5(content).hexdigest()
+            if self.sanitize_fs.exists({'md5': md5_hash, 'filename': file_name}):
+                return  # there is already an up to date entry -> do nothing
+            for old_entry in self.sanitize_fs.find({'filename': file_name}):  # delete old entries first
```

Comment:
Is there a query filter for GridFS? Like can you only request the `_id` such as is possible with regular MongoDB queries.

Reply:
It seems like there is no "projection" parameter for the `find()` method in GridFS.